source/val/validate_annotation.cpp

```diff
@@ -143,6 +143,11 @@ spv_result_t ValidateDecorationTarget(ValidationState_t& _, spv::Decoration dec,
         return fail(0) << "must be a variable";
       }
       break;
+    case spv::Decoration::PerPrimitiveNV:
+      if (target->opcode() != spv::Op::OpVariable) {
+        return fail(0) << "must be a memory object declaration";
```

Comment:
Memory object declarations include OpFunctionParameter (and are handled in the next block of decorations). The spec matches that, is this a bug here or in the extension?